app/controllers/tag_controller.rb

```diff
@@ -321,7 +321,7 @@ def contributors
                  .references(:term_data, :node_revisions)
                  .where('term_data.name = ?', params[:id])
                  .order('node_revisions.timestamp DESC')
-    @users = @notes.collect(&:author).uniq
+    @users = DrupalUser.where(uid: Tag.contributors(@tagnames[0]))
```

Comment:
Is it possible to use `User.where(id: ...` here instead, since we're trying to stop using DrupalUsers? Thank you! This otherwise looks fantastic.

Reply:
Hi, I guess we'd have to refactor the Tag.contributors method then, which currently  gives  array of uid (drupal_user id)!! Let's give it a try.

Reply:
@jywarren I guess we need to write a conversion function in user mode which fetches user_ids against drupal_user_ids. Then we could replace the DrupalUser with User in L324. Do you think this is the correct way, however it doesn't completely rules out DrupalUser.
The function would look something like (Would go in User Model):-
```
def self.user_ids(drupal_user_ids)
  user_ids=[]
  drupal_users = DrupalUser.include(:user).where(uid: drupal_user_ids)
  drupal_users.each do |du|
    user_ids<<du.user.id
  end
end
```
This would greatly increase the overload time since a separate query is required to fetch a user from drupal_user. Do you think we should do this???

Reply:
So, we've synced `User.id` and `DrupalUser.uid` and you can treat one like the other -- we did a big database migration a while back to ensure this. Does that help?